onnxruntime/test/framework/shape_inference_test.cc

```diff
@@ -76,6 +78,109 @@ TEST_F(ShapeInferenceTest, BasicTest) {
   CheckShapeEquality(InputShape(node), OutputShape(node));
 }
 
+TEST(ShapeInferenceV2Test, PartialDataPropagationTest) {
+  {
+    // Model #1
+    // This model contains "Shape" and "Reshape" operators.
+    auto model_path = ORT_TSTR("testdata/test_shape_data_propagation_with_shape_related_nodes.onnx");
+
+    Ort::SessionOptions session_options{};
+    session_options.SetGraphOptimizationLevel(ORT_DISABLE_ALL);
+
+    const OrtApi* g_ort = OrtGetApiBase()->GetApi(ORT_API_VERSION);
+    ORT_ENFORCE(g_ort->AddFreeDimensionOverrideByName(session_options, "batch", 1) == nullptr);
```

Comment:
This should be C++

Reply:
It seems we don't have a C++ wrapper for `AddFreeDimensionOverrideByName()` C API.

Reply:
Let's add one and this test would provider coverage.

Reply:
It is just a few lines of code, why not here?